rule_packages/c/Declarations6.json

```diff
@@ -0,0 +1,143 @@
+{
+  "MISRA-C-2012": {
+    "RULE-17-3": {
+      "properties": {
+        "obligation": "mandatory"
+      },
+      "queries": [
+        {
+          "description": "Omission of type specifiers may not be supported by some compilers. Additionally implicit typing can lead to undefined behaviour.",
+          "kind": "problem",
+          "name": "A function shall not be declared implicitly",
+          "precision": "very-high",
+          "severity": "error",
+          "short_name": "FunctionDeclaredImplicitly",
+          "tags": [
+            "correctness",
+            "readability"
+          ]
+        }
+      ],
+      "title": "A function shall not be declared implicitly"
+    },
+    "RULE-18-7": {
+      "properties": {
+        "obligation": "required"
+      },
+      "queries": [
+        {
+          "description": "The use of flexible array members can lead to unexpected program behaviour.",
+          "kind": "problem",
+          "name": "Flexible array members shall not be declared",
+          "precision": "very-high",
+          "severity": "error",
+          "short_name": "FlexibleArrayMembersDeclared",
+          "tags": [
+            "correctness"
+          ]
+        }
+      ],
+      "title": "Flexible array members shall not be declared"
+    },
+    "RULE-5-8": {
+      "properties": {
+        "obligation": "required"
+      },
+      "queries": [
+        {
+          "description": "Using non-unique identifiers can lead to developer confusion.",
+          "kind": "problem",
+          "name": "Identifiers that define objects or functions with external linkage shall be unique",
+          "precision": "very-high",
+          "severity": "error",
+          "short_name": "IdentifiersWithExternalLinkageNotUnique",
+          "tags": [
+            "maintainability",
+            "readability"
+          ]
+        }
+      ],
+      "title": "Identifiers that define objects or functions with external linkage shall be unique"
+    },
+    "RULE-5-9": {
+      "properties": {
+        "obligation": "advisory"
+      },
+      "queries": [
+        {
+          "description": "Using non-unique identifiers can lead to developer confusion.",
+          "kind": "problem",
+          "name": "Identifiers that define objects or functions with internal linkage should be unique",
+          "precision": "very-high",
+          "severity": "error",
+          "short_name": "IdentifiersWithInternalLinkageNotUnique",
+          "tags": [
+            "maintainability",
+            "readability"
+          ]
+        }
+      ],
+      "title": "Identifiers that define objects or functions with internal linkage should be unique"
```

Comment:
I would document the omission of detection of non-unique (multiple translation units) inline functions defined in a header file from the query's implementation in implementation_scope.